### PR TITLE
feat(validation): add validate mojo

### PR DESCRIPTION
**Pull request by akantcheff**

### CODE REVIEW

File: plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            executeXmlValidation(applicationXsdPath, applicationSourceDir);
+
+            log.info("Executing XML validation on Profiles");
+            executeXmlValidation(profileXsdPath, profileSourceDir);
+
+            log.info("Executing XML validation on BDM");
+            executeXmlValidation(bdmXsdPath, bdmSourceDir, BDM_SOURCE_FILE_REGEX);
+
+            log.info("Executing XML validation on BDM Access Control");
+            // FIXME remove this try-catch after extracting BDM Access Control model from bonita-common-sp
```
Comment (akantcheff) [inline plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java:91]: FIXME

File: plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            try {
+                executeXmlValidation(bdmAccessControlXsdPath, bdmAccessControlSourceDir,
+                        BDM_ACCESS_CONTROL_SOURCE_FILE_REGEX);
+            } catch (ValidationErrorException e) {
+                log.warn(bdmAccessControlXsdPath + " is not available in the classpath", e);
+            }
+
+            log.info("Executing XML validation on Organizations");
+            // FIXME remove this try-catch after extracting Organization model from bonita-process-engine
```
Comment (akantcheff) [inline plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java:100]: FIXME

File: plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            try {
+                executeXmlValidation(organizationXsdPath, organizationSourceDir, ORGANIZATION_SOURCE_FILE_REGEX);
+            } catch (ValidationErrorException e) {
+                log.warn(organizationXsdPath + " is not available in the classpath", e);
+            }
+
+            // TODO uid validation
```
Comment (akantcheff) [inline plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java:107]: TODO

File: plugin/pom.xml [hunk 4]
```diff
@@ -193,6 +193,12 @@
       <artifactId>logback-classic</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+      <scope>provided</scope>
+      <optional>true</optional>
```
Comment (rbioteau) [inline plugin/pom.xml:200]:
```suggestion
```

File: plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+ * <ul>
+ * <li>Applications</li>
+ * <li>Profiles</li>
+ * <li>BDM</li>
+ * <li>BDM Access Control</li>
+ * <li>Organizations</li>
+ * </ul>
+ */
+@Slf4j
+@Mojo(name = "validate", defaultPhase = LifecyclePhase.NONE)
```
Comment (akantcheff) [inline plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java:43]: Why don't we set the default phase to `VALIDATE` ?

Comment (akantcheff) [reply plugin/src/main/java/org/bonitasoft/plugin/validation/ValidateMojo.java]: Set to VALIDATE

**Review by rbioteau — APPROVED** [review]

**Pull request merged**